osx/KA-Lite/KA-Lite/AppDelegate.m

```diff
@@ -388,60 +412,14 @@ - (BOOL)string:(NSString *)string containsAllCharactersInString:(NSString *)char
 }
 
 
-//<##>runRootCommands
-BOOL runRootCommands(NSString *command) {
-//    NSString *msg = [NSString stringWithFormat:@"Running root command/s: %@...", command];
-//    showNotification(msg);
-    
-    NSDictionary *errorInfo = runAsRoot(command);
-    if (errorInfo != nil) {
-//        msg = [NSString stringWithFormat:@"FAILED command/s %@ with ERROR: %@", command, errorInfo];
-//        showNotification(msg);
-        return FALSE;
-    }
-//    msg = [NSString stringWithFormat:@"Done running root command/s %@.", command];
-//    showNotification(msg);
-    return TRUE;
-}
-
-
-//<##>setLaunchAgent
-NSString *getLaunchAgentCommand(NSString *source, NSString *target) {
-    if (pathExists(source)) {
-        return [NSString stringWithFormat:@"cp '%@' '%@'", source, target];
-    }
-    return nil;
-}
-
-
-// Not used for now but is useful to re-run the command individually later.
-BOOL setLaunchAgent(NSString *source, NSString *target) {
-    // Needs to run as root.
-    NSString *msg;
-    if (pathExists(source)) {
-        msg = [NSString stringWithFormat:@"Copying %@ to %@...", source, target];
-        showNotification(msg);
-        
-        NSString *command = [NSString stringWithFormat:@"cp '%@' '%@'", source, target];
-        NSDictionary *errorInfo = runAsRoot(command);
-        if (errorInfo != nil) {
-            msg = [NSString stringWithFormat:@"FAILED command %@ with ERROR: %@", command, errorInfo];
-            showNotification(msg);
-            return FALSE;
-        }
-        msg = [NSString stringWithFormat:@"Done copying %@ to %@.", source, target];
-        showNotification(msg);
-    } else {
-        msg = [NSString stringWithFormat:@"Source %@ OR target: %@ does not exist!", source, target];
-        showNotification(msg);
-        return FALSE;
-    }
-    return TRUE;
+NSString *getUsrBinKalite() {
+    return @"/usr/bin/kalite";
 }
 
-
-NSString *getUsrBinKalite() {
-    return @"/usr/local/bin/kalite";
+NSString *getCustomKaliteHomePath() {
+    NSUserDefaults *prefs = [NSUserDefaults standardUserDefaults];
+    NSString *kaliteHomePath = [prefs stringForKey:@"customKaliteData"];
+    return kaliteHomePath;
```

Comment:
Pls check if the path exists here and default to `~/.kalite/` because you are re-using this function in `setEnvVars()` and others.  There's a possibility that the user preferences were not yet set so it'll return an empty string if you don't have a default.